pkg/gencosmosdb/cosmosdb/cosmosdb.go

```diff
@@ -73,16 +74,39 @@ func (c *databaseClient) authorizeRequest(req *http.Request, resourceType, resou
 }
 
 func (c *databaseClient) do(ctx context.Context, method, path, resourceType, resourceLink string, expectedStatusCode int, in, out interface{}, headers http.Header) error {
+	for retry := 1; retry <= c.maxRetries; retry++ {
+		resp, rErr := c._do(ctx, method, path, resourceType, resourceLink, expectedStatusCode, in, out, headers)
```

Comment:
keep hold of this error so that you can return it verbatim on line 93 instead of generating a synthetic error